### PR TITLE
Adding all env variable groups

### PR DESCRIPTION
**Pull request by adusumillipraveen**
Adding all AKS env variable groups.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: variablegroups/DEMO-AKS-KeyVault.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      "isSecret": true
+    }
+  },
+  "id": 21,
+  "type": "AzureKeyVault",
+  "name": "DEMO-AKS-KeyVault",
+  "providerData": {
+    "serviceEndpointId": "420fac9f-d4bc-4c3f-8f70-e5283bdaaef5",
+    "vault": "cftapps-demo",
+    "lastRefreshedOn": "2019-07-03T13:24:20.107Z"
```
Comment (timja) [inline variablegroups/DEMO-AKS-KeyVault.json]: ?

File: variablegroups/DEMO-AKS-KeyVault.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "contentType": "",
+      "value": null,
+      "isSecret": true
+    }
+  },
+  "id": 21,
+  "type": "AzureKeyVault",
+  "name": "DEMO-AKS-KeyVault",
+  "providerData": {
+    "serviceEndpointId": "420fac9f-d4bc-4c3f-8f70-e5283bdaaef5",
```
Comment (timja) [inline variablegroups/DEMO-AKS-KeyVault.json]: this is a bit annoying 😢

File: variablegroups/scripts/get-all-variable-groups.sh [hunk 3]
```diff
@@ -10,11 +10,11 @@ fi
 ACCESSTOKEN=$1
 USERNAME=${2:azuredevops}
 
-groupList=$(curl -X GET -s -u ${USERNAME}:${ACCESSTOKEN} "https://dev.azure.com/hmcts/cnp/_apis/distributedtask/variablegroups/?api-version=5.0-preview.1" | jq -r '.value[] | .name' )
+groupList=$(curl -X GET -s -u ${USERNAME}:${ACCESSTOKEN} "https://dev.azure.com/hmcts/cnp/_apis/distributedtask/variablegroups/?api-version=5.0-preview.1" | jq -r '.value[] | .name  | if contains("AKS") then . else empty end' )
 
 
 for group in ${groupList} ; do
 
-curl -X GET -s -u ${USERNAME}:${ACCESSTOKEN} "https://dev.azure.com/hmcts/cnp/_apis/distributedtask/variablegroups/?groupName=${group}&api-version=5.0-preview.1" | jq -r '.value[0] | del(.createdOn, .modifiedBy , .createdBy, .modifiedOn, .providerData.lastRefreshedOn)' > ${group}.json
```
Comment (timja) [inline variablegroups/scripts/get-all-variable-groups.sh]: this didn't work?

Comment (adusumillipraveen) [reply variablegroups/scripts/get-all-variable-groups.sh]: yes, this doesn't work . It leaves it in non editable stage.

File: variablegroups/DEMO-AKS-KeyVault.json (hunk 2 above)
Comment (adusumillipraveen) [reply variablegroups/DEMO-AKS-KeyVault.json]: yeah, i actually wanted to get rid of using keyvaults.  Raised this a while ago https://dev.azure.com/hmcts/Platform%20Engineering/_workitems/edit/306/ .
But, i will write what to do to get that id in readme.

File: variablegroups/RPE-AKS-Common.json [hunk 4]
```diff
@@ -1,59 +0,0 @@
-{
```
Comment (timja) [inline variablegroups/RPE-AKS-Common.json]: is it worth keeping these so we can setup an RPE tenant env easily?

File: README.md [hunk 5]
```diff
@@ -34,3 +34,14 @@ ARM|Add Default Metric Alerts|Y|N|
 ARM|Add Custom Metric Alerts|Y|N|
 
 
+## Usage Guidelines
+
+### Variable Groups
+
+As maintaining Variable groups in UI is tedious , all AKS pipeline variable groups are managed as code in here.
```
Comment (timja) [inline README.md]:
```suggestion
As maintaining Variable groups in UI is tedious, all AKS pipeline variable groups are managed as code in here.
```

File: README.md [hunk 6]
```diff
@@ -34,3 +34,14 @@ ARM|Add Default Metric Alerts|Y|N|
...
+
+* To update repository with variable groups if changed through UI , run [Get All Script][variablegroups/scripts/get-all-variable-groups.sh] with a Personal Access Token.
+* Pipeline uses [Update Script][variablegroups/scripts/update-variable-groups.sh] to update all variable groups in [Variable Groups][variablegroups/] while building master branch.
+* To create a new environment, run [Create Script][variablegroups/scripts/add-variable-groups.sh] with access token and environment name to create new variable groups. Fill all fields in the generated files with appropriate values before raising a PR.
```
Comment (timja) [inline README.md]: I don't quite understand the point to this, wouldn't it be simpler to just copy the files in source control? then you don't need a PAT locally?

File: README.md [hunk 7]
```diff
@@ -34,3 +34,14 @@ ARM|Add Default Metric Alerts|Y|N|
 ARM|Add Custom Metric Alerts|Y|N|
 
 
+## Usage Guidelines
+
+### Variable Groups
+
+As maintaining Variable groups in UI is tedious , all AKS pipeline variable groups are managed as code in here.
+
+* To update repository with variable groups if changed through UI , run [Get All Script][variablegroups/scripts/get-all-variable-groups.sh] with a Personal Access Token.
```
Comment (timja) [inline README.md]:
```suggestion
* To update repository with variable groups if changed through UI, run [Get All Script][variablegroups/scripts/get-all-variable-groups.sh] with a Personal Access Token.
```

Comment (timja) [reply README.md]: maybe worth pointing to the docs for personal access token?

File: variablegroups/RPE-AKS-Common.json (hunk 4 above)
Comment (adusumillipraveen) [reply variablegroups/RPE-AKS-Common.json]: I am not sure if this is being used and up to date . Especially didn't want to check in something which isn't working and tested . Does it make sense ?

File: README.md (hunk 6 above)
Comment (adusumillipraveen) [reply README.md]: I guess one should still want it to run get-all-variable-groups (This should be their personal token though) . But yes, it makes absolutely no sense to use this :) Unlike how i felt last night :)

File: README.md [hunk 8]
```diff
@@ -42,6 +42,7 @@ As maintaining Variable groups in UI is tedious, all AKS pipeline variable group
...
-* To create a new environment, run [Create Script][variablegroups/scripts/add-variable-groups.sh] with access token and environment name to create new variable groups. Fill all fields in the generated files with appropriate values before raising a PR.
-* You can get serviceEndpointId  for keyvaults using [Rest API][https://docs.microsoft.com/en-us/rest/api/azure/devops/serviceendpoint/endpoints/get%20service%20endpoints?view=azure-devops-rest-5.0] 
+* To create a new environment, copy and rename [SBOX-AKS-COMMON.json][variablegroups/SBOX-AKS-COMMON.json] ,[SBOX-AKS-Tags.json][variablegroups/SBOX-AKS-Tags.json], [SBOX-AKS-KeyVault.json][variablegroups/SBOX-AKS-KeyVault.json] . Fill all fields with appropriate values before raising a PR.
+* You can get serviceEndpointId  for KeyVault Variable Group using [Rest API][https://docs.microsoft.com/en-us/rest/api/azure/devops/serviceendpoint/endpoints/get%20service%20endpoints?view=azure-devops-rest-5.0] . If you don't have necessary privileges to list service connections, ask someone who has it.
```
Comment (timja) [inline README.md]: could you add a script that takes the name of the service connection and returns the ID?

File: variablegroups/RPE-AKS-Common.json (hunk 4 above)
Comment (adusumillipraveen) [reply variablegroups/RPE-AKS-Common.json]: Added them back.

**Pull request merged**